src/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs

```diff
@@ -94,5 +98,40 @@ public long ExtendedExpiresIn
 
         public DateTimeOffset AccessTokenExpiresOn { get; private set; }
         public DateTimeOffset AccessTokenExtendedExpiresOn { get; private set; }
+
+        public string Authority { get; private set; }
+
+        internal static MsalTokenResponse CreateFromBrokerResponse(Dictionary<string, string> responseDictionary)
```

Comment:
>static [](start = 17, length = 6)

I tried not making this static and had lots of issues. fyi for Henry. to get unblocked and moving on, i made it static. \. ./

Reply:
essentially this is a factory method. given these are really data classes you should consider creating another class and file with this helper method

---
In reply to: [256603323](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/881#discussion_r256603323) [](ancestors = 256603323)